api/desecapi/tests/testdomains.py

```diff
@@ -177,13 +196,48 @@ def testDomainDetailURL(self):
 
 class AuthenticatedDynDomainTests(APITestCase):
     def setUp(self):
+        httpretty.reset()
+        httpretty.disable()
         if not hasattr(self, 'owner'):
             self.owner = utils.createUser(dyn=True)
             self.ownedDomains = [utils.createDomain(self.owner, dyn=True), utils.createDomain(self.owner, dyn=True)]
             self.otherDomains = [utils.createDomain(), utils.createDomain()]
             self.token = utils.createToken(user=self.owner)
             self.client.credentials(HTTP_AUTHORIZATION='Token ' + self.token)
 
+    def testCanDeleteOwnedDynDomain(self):
+        httpretty.enable()
+        httpretty.register_uri(httpretty.DELETE, settings.NSLORD_PDNS_API + '/zones/' + self.ownedDomains[1].name + '.')
+        httpretty.register_uri(httpretty.DELETE, settings.NSMASTER_PDNS_API + '/zones/' + self.ownedDomains[1].name+ '.')
+        httpretty.register_uri(httpretty.PATCH, settings.NSLORD_PDNS_API + '/zones/dedyn.io.')
+
+        url = reverse('domain-detail', args=(self.ownedDomains[1].pk,))
+        response = self.client.delete(url)
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertEqual(httpretty.last_request().method, 'PATCH')
+        self.assertEqual(httpretty.last_request().headers['Host'], 'nslord:8081')
+        self.assertTrue('"NS"' in httpretty.last_request().parsed_body
+                        and '"' + self.ownedDomains[1].name + '."' in httpretty.last_request().parsed_body
+                        and '"DELETE"' in httpretty.last_request().parsed_body)
+
+        httpretty.reset()
+        httpretty.register_uri(httpretty.DELETE, settings.NSLORD_PDNS_API + '/zones/' + self.ownedDomains[1].name + '.')
+        httpretty.register_uri(httpretty.DELETE, settings.NSMASTER_PDNS_API + '/zones/' + self.ownedDomains[1].name+ '.')
+
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertTrue(isinstance(httpretty.last_request(), httpretty.core.HTTPrettyRequestEmpty))
+
+    def testCantDeleteOtherDynDomains(self):
+        httpretty.enable()
+        httpretty.register_uri(httpretty.DELETE, settings.NSLORD_PDNS_API + '/zones/' + self.otherDomains[1].name + '.')
+        httpretty.register_uri(httpretty.DELETE, settings.NSMASTER_PDNS_API + '/zones/' + self.otherDomains[1].name+ '.')
+
+        url = reverse('domain-detail', args=(self.otherDomains[1].pk,))
+        response = self.client.delete(url)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
```

Comment:
If 200 should be returned for non-existing domains, should be also return that for domains the user does not have access to? Or should we return 404 to increase privacy? But isn't the domain data public anyways? (So it should be 403?)